notebooks/simple_light_curve.jl

```diff
@@ -0,0 +1,179 @@
+
+# ╔═╡ 65a7b174-70c4-11eb-12bb-1fbc53ee1764
+md"""
+We can now use this as input into a limb darkening law, which we define next:
+"""
+
+# ╔═╡ e7223d2a-70c7-11eb-24e4-af0ff49d42db
+u = [0.4, 0.26] # Quadratic limb darkening
+
+# ╔═╡ 570e5624-70b8-11eb-1767-6911283302f5
+ld = PolynomialLimbDark(u)
+
+# ╔═╡ 40820d3a-70c5-11eb-04d1-3f32d0d453fe
+md"""
+For this example, we are using a Polynomial limb darkening law from [Agol, Luger, Foreman-Mackey (2020)](https://ui.adsabs.harvard.edu/abs/2020AJ....159..123A/abstract). This law is part of the `AbstractLimbDark` supertype, which includes the following laws as well that we will explore in other notebooks:
+"""
+
+# ╔═╡ 832cd21e-70c5-11eb-05b5-cd9a588751fc
+subtypes(Transits.AbstractLimbDark)
+
+# ╔═╡ d0863d52-70c5-11eb-22f6-2fb13f25751c
+md"""
+The length of the vector of limb darkening coefficients we passed above is equivalent to the order of the polynomial used.
+"""
+
+# ╔═╡ 10065930-70c6-11eb-1fa7-551fd12f8e75
+md"## Light curve computation"
+
+# ╔═╡ 1fff2420-70c6-11eb-1bc0-9d35bd83a36a
+md"""
+With the orbit and limb darkening law defined, we can now compute light curves over time `t` for a range of different planet-to-star ratios `rprs`, which can be controlled with the slider below. We perform the computation of each light curve by directly passing our `AbstractOrbit` object to `ld`:
+"""
+
+# ╔═╡ 6514b6ea-1374-4852-a4a7-ab8fa496d5d0
+md"""
+``R_\text{p}/R_* =`` $(@bind rprs Slider(0:0.05:0.2, show_value=true))
+"""
+
+# ╔═╡ 8be23852-70b8-11eb-30d3-b315246efe02
+begin
+	
+	# Initialize plot
+	p = plot(
+		ylim = (0.95, 1.0),
+		xguide = "Time from mid-transit",
+		yguide = "Relative flux",
+		legendtitle = L"R_\mathrm{p}/R_{*}",
+		title = "u = $u",
+	)
+	
+	
+	# Range of times to compute light curves over
+	t = range(-1, 1, length=1_000)u"hr"
+	
+	# Show previous models
+	rprs_previous = 0.0:0.05:rprs
+	fluxes_previous = @. ld(orbit, t, rprs_previous')
+	plot!(p, t, fluxes_previous, c=:lightgrey, label=false)
+	
+	# Show current model
+	fluxes = @. ld(orbit, t, rprs)
+	plot!(t, fluxes, c=1, label=rprs, legend=:bottomleft)
+end
+
+# ╔═╡ a5989098-70c7-11eb-24d3-fb1c85b6f770
+md"""
+Not too bad!
+"""
+
+# ╔═╡ af07537e-70bf-11eb-3c0a-592a55780281
+note(text) = Markdown.MD(Markdown.Admonition("note", "Note", [text]))
+
+# ╔═╡ ee993496-70bd-11eb-3374-83209e8150de
+note(
+md"""
+These fields are described in the documentation, which can be readily pulled up in the Pluto notebook by selecting the `Live docs` tab below and then clicking on `SimpleOrbit` in the cell above. The `AbstractOrbit` supertype also inlcludes a `KeplerianOrbit` subtype that we will explore in the next notebook.
+"""
+)
+
+# ╔═╡ 0d935990-70c7-11eb-227a-274d8776916d
+note(
+md"""
+Technically we are passing this to the `compute` function, which is aliased to `ld` for convenience. A row-vector of planet-to-star ratios can be passed as well, and the required light curve computations will be automatically broadcasted:
```

Comment:
```suggestion
md"""
All of the limb darkening laws can be called like a function `ld(b, r)` or using the `compute` method- `compute(ld, b, r)`. We provide both for convenience- it's natural and easy to treat the laws as functions but some code is easier to write using the concrete `compute` method. In addition, we don't provide any default vectorized versions of the methods, since Julia does not need to be vectorized for speed (unlike numpy) this lets users apply our laws in a variety of ways (e.g., `map`, `broadcast`, `Threads.@threads`, etc.). Below we use [broadcasting](https://docs.julialang.org/en/v1/manual/arrays/#Broadcasting) with the `@.` macro to create a grid of outputs across the `t` and `rprs` column and row vectors.
```

Want to avoid causing undue confusion between `compute` and the "functor" style. Also, might be a little more gentle introducing broadcasting- for people coming from python they'll have no idea why everything doesn't just vectorize automatically.

Reply:
done